### PR TITLE
Fix modulefiles for Hera/Rocky8 OS.

### LINKED ISSUE #2167: Need new modules for ufs build on hera rocky nodes, jet rocky nodes, and orion rocky nodes

### PR DESCRIPTION
**Pull request by RatkoVasic-NOAA**
## Commit Queue Requirements:
- [x] Fill out all sections of this template.
- [x] All sub component pull requests have been reviewed by their code managers.
- [ ] Run the full Intel+GNU RT suite (compared to current baselines) on either Hera/Derecho/Hercules
- [ ] Commit 'test_changes.list' from previous step
---
## Description:
Hera is switching to new OS. This is update to enable ufs-weather-model to run on Rocky8 OS.
Necessary changes are made to spack-stack libraries.
**NOTE! Since different version of openmpi is used, results change when using GNU compiler.**

### Commit Message:

- update module paths in Hera intel/gnu lua files for Rocky8 OS spack-stack libraries.

### Priority:
* Normal

### UFSWM:
* Closes #2167 and #2143 

### Sub component Pull Requests:
* None

### UFSWM Blocking Dependencies:
* None

---
### Regression Test Changes (Please commit test_changes.list):
* PR Updates/Changes Baselines. (Only GNU is expected to change results)

### Input data Changes:
* None.

### Library Changes/Upgrades:
Library changes are included in this PR (spack-stack).
  
---

## Testing Log:
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Hercules
  - [ ] Jet
  - [x] Gaea
  - [x] Derecho
- WCOSS2
  - [ ] Dogwood/Cactus
  - [ ] Acorn
- [ ] CI
- [ ] opnReqTest (complete task if unnecessary)

### CODE REVIEW

**Comment by RatkoVasic-NOAA** [comment]
This should be tested on Rocky login nodes! hfe09-hfe12.

**Comment by junwang-noaa** [comment]
@RatkoVasic-NOAA With this module update, we can only run UFS weather model code on hfe09-hfe12, can we run the model on other hera nodes when your PR is committed? Thanks

**Comment by RatkoVasic-NOAA** [comment]
> @RatkoVasic-NOAA With this module update, we can only run UFS weather model code on hfe09-hfe12, can we run the model on other hera nodes when your PR is committed? Thanks

@junwang-noaa , no. You cannot use both. But, you can use old modulefiles (`ufs_hera.gnu.lua  ufs_hera.intel.lua`), which are in use now. More and more resources are moved from CentOS to Rocky8, so it might be good to switch sooner than later.

**Comment by jkbk2004** [comment]
@RatkoVasic-NOAA can you continue to sync up branch? we may need to schedule this pr tomorrow.

**Comment by RatkoVasic-NOAA** [comment]
> @RatkoVasic-NOAA can you continue to sync up branch? we may need to schedule this pr tomorrow.

Done.

**Comment by SamuelTrahanNOAA** [comment]
The `control_wam_debug_gnu` test failed for me with a floating-point exception. I haven't tried resubmitting yet.

It is here:

HERA: `/scratch1/NCEPDEV/stmp2/Samuel.Trahan/FV3_RT/rt_3224016/control_wam_debug_gnu`

<details> <summary>Big long backtrace</summary>

```
srun: lua: This job was submitted from a host running Rocky 8. Assigning job to el8 reservation.
 21: 
 21: Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation.
 21: 
 21: Backtrace for this error:
 45: 
 45: Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation.
 45: 
 45: Backtrace for this error:
 21: #0  0x14c10f100b4f in ???
 21: #1  0x1ce24ec in __nh_utils_mod_MOD_sim1_solver
 21:    at /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/nested-workflow/sorc/ratko-rocky/FV3/atmos_cubed_sphere/model/nh_utils.F90:1557
 21: #2  0x1d0f9c7 in __nh_utils_mod_MOD_riem_solver_c._omp_fn.0
 21:    at /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/nested-workflow/sorc/ratko-rocky/FV3/atmos_cubed_sphere/model/nh_utils.F90:481
 21: #3  0x14c111078721 in GOMP_parallel
 21:    at /tmp/role.apps/spack-stage/spack-stage-gcc-9.2.0-ku6r4f5qa5obpfnqpa6pezhogxq6sp7h/spack-src/libgomp/parallel.c:171
 21: #4  0x1d02ae4 in __nh_utils_mod_MOD_riem_solver_c
 21:    at /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/nested-workflow/sorc/ratko-rocky/FV3/atmos_cubed_sphere/model/nh_utils.F90:501
 21: #5  0x18392b8 in __dyn_core_mod_MOD_dyn_core
 21:    at /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/nested-workflow/sorc/ratko-rocky/FV3/atmos_cubed_sphere/model/dyn_core.F90:636
 21: #6  0x18e743f in __fv_dynamics_mod_MOD_fv_dynamics
 21:    at /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/nested-workflow/sorc/ratko-rocky/FV3/atmos_cubed_sphere/model/fv_dynamics.F90:691
 21: #7  0x178590a in __atmosphere_mod_MOD_atmosphere_dynamics
 21:    at /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/nested-workflow/sorc/ratko-rocky/FV3/atmos_cubed_sphere/driver/fvGFS/atmosphere.F90:699
 21: #8  0x146d8c4 in __atmos_model_mod_MOD_update_atmos_model_dynamics
 21:    at /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/nested-workflow/sorc/ratko-rocky/FV3/atmos_model.F90:854
 21: #9  0x12cde44 in fcst_run_phase_1
 21:    at /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/nested-workflow/sorc/ratko-rocky/FV3/module_fcst_grid_comp.F90:1306
 21: #10  0x8dfb40 in ???
 21: #11  0x8dfeb4 in ???
 21: #12  0x7ee006 in ???
 21: #13  0x7f0ed8 in ???
 21: #14  0xde6522 in ???
 21: #15  0x8dde82 in ???
 21: #16  0x80c08c in ???
 21: #17  0xad748e in ???
 21: #18  0x12bbbea in modeladvance_phase1
 21:    at /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/nested-workflow/sorc/ratko-rocky/FV3/fv3_cap.F90:1026
 21: #19  0x12bbf34 in modeladvance
 21:    at /scratch2/BMC/wrfruc/Samuel.Trahan/westwater/nested-workflow/sorc/ratko-rocky/FV3/fv3_cap.F90:975
```

</details>

**Comment by SamuelTrahanNOAA** [comment]
The cpld_control_p8_gnu and cpld_debug_p8_gnu both fail with this message:

```
 21: --------------------------------------------------------------------------
 21: The OSC pt2pt component does not support MPI_THREAD_MULTIPLE in this release.
 21: Workarounds are to run on a single node, or to use a system with an RDMA
 21: capable network such as Infiniband.
 21: --------------------------------------------------------------------------
```

**Comment by jkbk2004** [comment]
[RegressionTests_hera.log](https://github.com/ufs-community/ufs-weather-model/files/14666372/RegressionTests_hera.log)

**Comment by jkbk2004** [comment]
> /scratch1/NCEPDEV/stmp2/Samuel.Trahan/FV3_RT/rt_3224016/control_wam_debug_gnu

@SamuelTrahanNOAA all tests pass on my side. @zach1221 @FernandoAndrade-NOAA can you test gnu cases on hera/rocky8?

**Comment by SamuelTrahanNOAA** [comment]
Did your tests pass on the first try or did you have to rerun them?

**Comment by jkbk2004** [comment]
> Did your tests pass on the first try or did you have to rerun them?

It passed with the first try.  A few other people are running gnu cases now. We can confirm.

**Comment by zach1221** [comment]
> The cpld_control_p8_gnu and cpld_debug_p8_gnu both fail with this message:
> 
> ```
>  21: --------------------------------------------------------------------------
>  21: The OSC pt2pt component does not support MPI_THREAD_MULTIPLE in this release.
>  21: Workarounds are to run on a single node, or to use a system with an RDMA
>  21: capable network such as Infiniband.
>  21: --------------------------------------------------------------------------
> ```

I am also receiving this error on Hera, for cpld_control_p8_gnu and cpld_debug_p8_gnu.

**Comment by jkbk2004** [comment]
@RatkoVasic-NOAA it sounds like different result with case-by-case. Some nodes still heterogeneous? openmpi or gcc version issue?

**Comment by SamuelTrahanNOAA** [comment]
@zach1221 - Can you reproduce the error I saw with control_wam_debug_gnu?

It may have been caused by the job being sent to the wrong service (login) node.

**Comment by jkbk2004** [comment]
I am not sure if we are triggering -mcmodel=medium on hera/gnu.

**Comment by BrianCurtis-NOAA** [comment]
For ecflow, if used: I have a feeling maybe the ECF_HOST env var on hera isn't set properly with this transition? I logged into a rocky8 node and 'module load ecflow' and 'printenv | grep ECF' and only the _ROOT env var showed up. Try manually setting the hera ecflow ECF_HOST var to (i think) hfe12 and see if that helps? (if needed)

**Comment by zach1221** [comment]
> @zach1221 - Can you reproduce the error I saw with control_wam_debug_gnu?
> 
> It may have been caused by the job being sent to the wrong service (login) node.

With my first test, where cpld_control_p8_gnu and cpld_debug_p8_gnu failed, control_wam_debug_gnu actually passed. I'm retesting now with some changes to cmake.gnu .

**Comment by SamuelTrahanNOAA** [comment]
I used Rocoto and saw those bugs. That means the problem is not specific to ecFlow.

**Comment by jkbk2004** [comment]
@climbfuji I am not sure about OSC pt2pt issue. I vaguely remember a similar issue was seen with openmpi on Hercules. Do you remember? @RatkoVasic-NOAA @ulmononian Any comment?

**Comment by RatkoVasic-NOAA** [comment]
@jkbk2004 I'm looking into this right now. I haven't seen this error message before.

**Comment by zach1221** [comment]
@jkbk2004 forcing to run on nodes 5-12 didnt work, failing with same OSC pt2pt error. The GNU.cmake update test timed out, so running it again with manually extended time.

Update: the cpld_control_p8_gnu test failed with same error after adding -mcmodel=large & medium to gnu.cmake.

**Comment by jkbk2004** [comment]
@zach1221 can you come with a list crashed cases?

**Comment by jkbk2004** [comment]
failed cases on hera: control_wam_debug_gnu, cpld_control_p8_gnu, and cpld_debug_p8_gnu. 
cpld_control_p8_gnu and pld_debug_p8_gnu dumps crashing err osc pt2pt. But control_wam_debug_gnu shows slightly different crashing behavior
```
 21: An MPI communication peer process has unexpectedly disconnected.  This
 21: usually indicates a failure in the peer process (e.g., a crash or
 21: otherwise exiting without calling MPI_FINALIZE first).
```
and PET000.ESMF_LogFile shows
```
nectedCplFields FV3 Import Field vtype is not connected
20240321 013524.316 INFO             PET000 /scratch2/NCEPDEV/marine/Jong.Kim/UFS-RT/pr-2194/FV3/cpl/module_cplfields.F90:545 realizeConnectedCplFields FV3 Import Field stype is not connected
20240321 013524.316 INFO             PET000 /scratch2/NCEPDEV/marine/Jong.Kim/UFS-RT/pr-2194/FV3/cpl/module_cplfields.F90:545 realizeConnectedCplFields FV3 Import Field vfrac is not connected
```
@DeniseWorthen can you take a look at /scratch1/NCEPDEV/stmp2/Jong.Kim/FV3_RT/pr_2194/control_wam_debug_gnu/PET000.ESMF_LogFile ?
@BrianCurtis-NOAA @junwang-noaa @zach1221 @FernandoAndrade-NOAA we may need to make a decision to turn off these 3 gnu cases and follow on with git issue to update gnu (or openmpi) verion. I mean if we reproduce all other intel and gnu cases. @RatkoVasic-NOAA FYI

**Comment by DeniseWorthen** [comment]
@jkbk2004 The control wam test is a standalone test, and I have no specialized knowledge about it. I can take a look later. 

However, w/rt to turning off GNU coupled tests on our only remaining GNU/cpld platform, my opinion is that it is acceptable only if we turn the cpld tests back *on* for Hercules at the same time.

**Comment by zach1221** [comment]
> @jkbk2004 The control wam test is a standalone test, and I have no specialized knowledge about it. I can take a look later.
> 
> However, w/rt to turning off GNU coupled tests on our only remaining GNU/cpld platform, my opinion is that it is acceptable only if we turn the cpld tests back _on_ for Hercules at the same time.

I can enable these on hercules and test them out.

Update: control_wam_debug gnu passed on hercules but cpld_control_p8_gnu, and cpld_debug_p8_gnu still fail, as Jong mentions on issue #2070.

**Comment by DeniseWorthen** [comment]
Prior to Tuesday's maintenance, had the GNU tests on Hera/R8 been tested and verified to be working?

**Comment by zach1221** [comment]
> Prior to Tuesday's maintenance, had the GNU tests on Hera/R8 been tested and verified to be working?

Yes, I believe Jong ran tests at some point yesterday on Hera, logs posted above, where these cpld tests were passing.

**Comment by jkbk2004** [comment]
My gut feeling is there could be something: OS kernel update with openmpi. Clearly, we need gnu version update.

**Comment by DeniseWorthen** [comment]
> > Prior to Tuesday's maintenance, had the GNU tests on Hera/R8 been tested and verified to be working?
> 
> Yes, I believe Jong ran tests at some point yesterday on Hera, logs posted above, where these cpld tests were passing.

But *prior* to the maintenance, were the cpld/GNU tests run ?

**Comment by climbfuji** [comment]
I do recall vaguely that on Hercules, the first openmpi installations from the sysadmins didn't work (same issues?), after enough complaining from me, they built a new version of openmpi 4.1.6 (5?) during a maintenance outage and that worked.

**Comment by jkbk2004** [comment]
> > > Prior to Tuesday's maintenance, had the GNU tests on Hera/R8 been tested and verified to be working?
> > 
> > 
> > Yes, I believe Jong ran tests at some point yesterday on Hera, logs posted above, where these cpld tests were passing.
> 
> But _prior_ to the maintenance, were the cpld/GNU tests run ?

Honestly, I am not so clear with previous tests.

**Comment by jkbk2004** [comment]
We need to update baseline with hera/rocy8. I am testing to see if the runs continuously reproduce.

**Comment by DeniseWorthen** [comment]
@jkbk2004 Isn't the issue that the cpld/GNU tests are not running on Hera/R8, period? Why are you now talking about updating baselines?

**Comment by RatkoVasic-NOAA** [comment]
Previously on CentOS7 we used JCSDA's installation of the openmpi: /scratch1/NCEPDEV/jcsda/jedipara/spack-stack/modulefiles/openmpi/4.1.5
Now, we use admin's installation of openmpi on Rocky8: openmpi/4.1.6_gnu9.2.0

**Comment by jkbk2004** [comment]
OS changes and new stack. It perfectly makes a sense to see results changes.

**Comment by DeniseWorthen** [comment]
@RatkoVasic-NOAA So when you were testing this PR, did you run w/ the SA's installation of openmpi? 

What I'm trying to figure out is when exactly we determined that cpld/GNU was not working for the Hera/R8 transition? We knew this was coming.

**Comment by RatkoVasic-NOAA** [comment]
@DeniseWorthen I haven't tested full rt.conf just part of it.

**Comment by RatkoVasic-NOAA** [comment]
As @climbfuji said. It might be with admins' installation of openmpi. Probably same thing happened before and we used JCSDA's installation instead?

**Comment by RatkoVasic-NOAA** [comment]
> @RatkoVasic-NOAA So when you were testing this PR, did you run w/ the SA's installation of openmpi?
> 
> What I'm trying to figure out is when exactly we determined that cpld/GNU was not working for the Hera/R8 transition? We knew this was coming.

@DeniseWorthen  With admins' installation.

**Comment by BrianCurtis-NOAA** [comment]
What version of GNU is on Hercules? Maybe we need to force the use of GNU 12/13 here , right now.

**Comment by DeniseWorthen** [comment]
> > @RatkoVasic-NOAA So when you were testing this PR, did you run w/ the SA's installation of openmpi?
> > What I'm trying to figure out is when exactly we determined that cpld/GNU was not working for the Hera/R8 transition? We knew this was coming.
> 
> @DeniseWorthen With admins' installation.

OK, so you tested w/ SA's openmpi, but you did not test the full rt.conf. Is that right?

**Comment by jkbk2004** [comment]
> What version of GNU is on Hercules? Maybe we need to force the use of GNU 12/13 here , right now.

@BrianCurtis-NOAA I agree, Syncing gnu version across platforms is good idea. hercules is gnu12.2.0.

**Comment by BrianCurtis-NOAA** [comment]
> > What version of GNU is on Hercules? Maybe we need to force the use of GNU 12/13 here , right now.
> 
> @BrianCurtis-NOAA I agree, Syncing gnu version across platforms is good idea. hercules is gnu12.2.0.

So how long would it take to get a spack-stack install using OpenMPI 4.1.6 and GNU 12.2 (or newer?)

**Comment by RatkoVasic-NOAA** [comment]
> > > @RatkoVasic-NOAA So when you were testing this PR, did you run w/ the SA's installation of openmpi?
> > > What I'm trying to figure out is when exactly we determined that cpld/GNU was not working for the Hera/R8 transition? We knew this was coming.
> > 
> > 
> > @DeniseWorthen With admins' installation.
> 
> OK, so you tested w/ SA's openmpi, but you did not test the full rt.conf. Is that right?

Right.

**Comment by BrianCurtis-NOAA** [comment]
> > > What version of GNU is on Hercules? Maybe we need to force the use of GNU 12/13 here , right now.
> > 
> > 
> > @BrianCurtis-NOAA I agree, Syncing gnu version across platforms is good idea. hercules is gnu12.2.0.
> 
> So how long would it take to get a spack-stack install using OpenMPI 4.1.6 and GNU 12.2 (or newer?)

They have 13.2.0 built with 4.1.6 it seems already?
```
------------------------------------------------------------------------------------------------------------------
  gnu:
------------------------------------------------------------------------------------------------------------------
     Versions:
        gnu/6.5.0
        gnu/9.2.0b
        gnu/9.2.0
        gnu/13.2.0


------------------------------------------------------------------------------------------------------------------
  openmpi:
------------------------------------------------------------------------------------------------------------------
     Versions:
        openmpi/3.1.4
        openmpi/4.1.6_gnu9.2.0
        openmpi/4.1.6_gnu13.2.0
        openmpi/4.1.6
```

**Comment by RatkoVasic-NOAA** [comment]
> 
> So how long would it take to get a spack-stack install using OpenMPI 4.1.6 and GNU 12.2 (or newer?)

@BrianCurtis-NOAA in may take anywhere from 12+ hours (providing everything goes ok), but that's not guarantee that it will work for all tests.
@DomHeinzeller as I understand we haven't tested spack-stack with gnu 13.2.0 yet, and no plans for near future?

**Comment by zach1221** [comment]
@jkbk2004 we've decided to proceed with testing on this PR, and disable the failing gnu cases for Hera and enable the one control_wam_debug_gnu on Hercules. I will create a new issue for the failing cases on Hera, and in the meantime @RatkoVasic-NOAA can work to install new spack-stack with gnu/13.2.0 and openmpi/4.16. on Hera. Given that may take some time, we will work only PRs that do not change baselines until new spack-stack installation is complete, and we can hopefully re-enable the Hera cpld gnu cases. @BrianCurtis-NOAA @DeniseWorthen please feel free to add anything I may have missed.

**Comment by climbfuji** [comment]
gnu@13 is very likely not going to work yet. I know for sure that packages like mapl didn't build, this was fixed only recently in the authoritative repo.

gnu@12.2 is recommended if you use it as the main compiler, but you can't use it as backend for the Intel compilers - that needs to be anything between 9.2 and 11.x

See https://github.com/JCSDA/spack-stack/blob/develop/configs/sites/discover-scu16/compilers.yaml where the default OS gnu compiler is too old and we have to inject a newer gcc compiler for C++-17 support into the Intel compiler config.

See https://github.com/JCSDA/spack-stack/blob/develop/configs/sites/hercules/compilers.yaml where this isn't necessary, because the default/OS GNU is 11.3.0.

On Hera, the default gcc is 8.5.0 - too old.

**Comment by RatkoVasic-NOAA** [comment]
> @jkbk2004 we've decided to proceed with testing on this PR, and disable the failing gnu cases for Hera and enable the one control_wam_debug_gnu on Hercules. I will create a new issue for the failing cases on Hera, and in the meantime @RatkoVasic-NOAA can work to install new spack-stack with gnu/13.2.0 and openmpi/4.16. on Hera. Given that may take some time, we will work only PRs that do not change baselines until new spack-stack installation is complete, and we can hopefully re-enable the Hera cpld gnu cases. @BrianCurtis-NOAA @DeniseWorthen please feel free to add anything I may have missed.

As @climbfuji explained, I'm not going to start working on GNU 13 until all packages are working with this version. Though, I will try on my personal space in the meantime.

**Comment by SamuelTrahanNOAA** [comment]
Do you have to use OpenMPI for this? Can't you use an MPICH derivative instead?

**Comment by climbfuji** [comment]
> Do you have to use OpenMPI for this? Can't you use an MPICH derivative instead?

If you want to use mapl@2.40.3 then you can't use mpich@4 - don't remember when the bug fix in mapl was merged that allows using mpich@4

**Comment by BrianCurtis-NOAA** [comment]
> gnu@13 is very likely not going to work yet. I know for sure that packages like mapl didn't build, this was fixed only recently in the authoritative repo.
> 
> gnu@12.2 is recommended if you use it as the main compiler, but you can't use it as backend for the Intel compilers - that needs to be anything between 9.2 and 11.x
> 
> See https://github.com/JCSDA/spack-stack/blob/develop/configs/sites/discover-scu16/compilers.yaml where the default OS gnu compiler is too old and we have to inject a newer gcc compiler for C++-17 support into the Intel compiler config.
> 
> See https://github.com/JCSDA/spack-stack/blob/develop/configs/sites/hercules/compilers.yaml where this isn't necessary, because the default/OS GNU is 11.3.0.
> 
> On Hera, the default gcc is 8.5.0 - too old.

Gnu 12.2 is fine with me. I wasn't sure how complicated it would make things on hera. Important to get this started ASAP.

**Comment by climbfuji** [comment]
> > Do you have to use OpenMPI for this? Can't you use an MPICH derivative instead?
> 
> If you want to use mapl@2.40.3 then you can't use mpich@4 - don't remember when the bug fix in mapl was merged that allows using mpich@4

mapl@2.42.0 works with mpich@4 - https://github.com/GEOS-ESM/MAPL/releases/tag/v2.42.0

**Comment by RatkoVasic-NOAA** [comment]
> 
> Gnu 12.2 is fine with me. I wasn't sure how complicated it would make things on hera. Important to get this started ASAP.

We don't have 12.2 on Hera/Rocky. Only 9.2.0 and 13.2.0 (for now).

**Comment by BrianCurtis-NOAA** [comment]
> > 
> > Gnu 12.2 is fine with me. I wasn't sure how complicated it would make things on hera. Important to get this started ASAP.
> 
> We don't have 12.2 on Hera/Rocky. Only 9.2.0 and 13.2.0 (for now).

How long would it take their SA to install 12.2? Would it be easier to wait for that over trying to get 13.2 working with spack stack?

**Comment by RatkoVasic-NOAA** [comment]
> > > Gnu 12.2 is fine with me. I wasn't sure how complicated it would make things on hera. Important to get this started ASAP.
> > 
> > 
> > We don't have 12.2 on Hera/Rocky. Only 9.2.0 and 13.2.0 (for now).
> 
> How long would it take their SA to install 12.2? Would it be easier to wait for that over trying to get 13.2 working with spack stack?

That is good question, meaning: I don't know the answer ;-) I will first try with 13.2.0 and see what we need only for WM.

**Comment by climbfuji** [comment]
> > > > Gnu 12.2 is fine with me. I wasn't sure how complicated it would make things on hera. Important to get this started ASAP.
> > > 
> > > 
> > > We don't have 12.2 on Hera/Rocky. Only 9.2.0 and 13.2.0 (for now).
> > 
> > 
> > How long would it take their SA to install 12.2? Would it be easier to wait for that over trying to get 13.2 working with spack stack?
> 
> That is good question, meaning: I don't know the answer ;-) I will first try with 13.2.0 and see what we need only for WM.

@RatkoVasic-NOAA It **won't work** since mapl doesn't work with 13.2 and you need that for the UFSWM. The last change for gnu@13 for mapl was apparently merged last week, there isn't even a release yet - https://github.com/GEOS-ESM/MAPL/pull/2640. - EDIT this was for mapl@3.

I don't know which tag if any of mapl@2 works with gnu@13 - @mathomp4 probably knows.

**Comment by mathomp4** [comment]
> > 
> > 
> > That is good question, meaning: I don't know the answer ;-) I will first try with 13.2.0 and see what we need only for WM.
> 
> @RatkoVasic-NOAA It **won't work** since mapl doesn't work with 13.2 and you need that for the UFSWM. The last change for gnu@13 for mapl was apparently merged last week, there isn't even a release yet - [GEOS-ESM/MAPL#2640](https://github.com/GEOS-ESM/MAPL/pull/2640). - EDIT this was for mapl@3.
> 
> I don't know which tag if any of mapl@2 works with gnu@13 - @mathomp4 probably knows.

At the moment no official release of MAPL 2 works with GCC 13. But, MAPL `develop` (and MAPL3)  does thanks to @tclune. It would require the [GFE v1.13 libraries](https://github.com/Goddard-Fortran-Ecosystem/GFE/releases/tag/v1.13.0) as those had workarounds for GCC 13.[^1] 

That said, if needed we could release MAPL 2.45 with those fixes...but not that at the moment MAPL 2.44+ doesn't build in spack. That is due to the `ESMF::ESMF` target business. ESMF has or will soon release a beta snapshot of 8.6.1 that should have the fixes for that in though some upstream repos will need to [update their `FindESMF.cmake` files](https://github.com/esmf-org/esmf/pull/226#issuecomment-1994662883) (though maybe not for spack? confusing sometimes)

[^1]: I am currently building Baselibs and GEOS to test GCC 13 and the full model. My guess is GEOSgcm should be fine as long as MAPL is. MAPL is where the fancy Fortran is.

**Comment by BrianCurtis-NOAA** [comment]
I didn't think we needed to run all systems. No code touches any other system. I thought Hercules was a special case because of the changes to the cpld tests.

**Comment by zach1221** [comment]
> I didn't think we needed to run all systems. No code touches any other system. I thought Hercules was a special case because of the changes to the cpld tests.

Ok, I was just being safe. We don't have to finish jet and wcoss2/acorn if you don't think it's necessary. But yes, you're correct, only hercules/hera had changes.

**Comment by zach1221** [comment]
@BrianCurtis-NOAA @DeniseWorthen @jkbk2004 testing is complete. Feel free to provide final review.

File: tests/rt.conf [hunk 1]
```diff
@@ -413,13 +413,13 @@ RUN | rap_control_dyn64_phy32_debug                     | + hera hercules
 
 ### S2S tests ###
 COMPILE | s2swa | gnu | -DAPP=S2SWA -DCCPP_SUITES=FV3_GFS_v17_coupled_p8 | + hera hercules | fv3 |
-RUN | cpld_control_p8                                   | + hera                     | baseline |
+# RUN | cpld_control_p8                                   | + hera                     | baseline |
```
Comment (DeniseWorthen) [inline tests/rt.conf:416]: I didn't realize this previously, but it should probably be noted that the two failed tests use both aerosols and waves. The non-wave,no-aerosol test *does* run on hera/R8 and the wave+pdlib runs. This indicates to me that the issue may be aerosol related? Did anyone try to run the cpld_debug_p8 but turn off aerosols?

Comment (zach1221) [reply tests/rt.conf:416]: I did not try this, but can give it a go now.

Comment (zach1221) [reply tests/rt.conf:416]: I have not been able to get this to pass yet. Getting a lot of timeouts now oddly.

Comment (jkbk2004) [reply tests/rt.conf:416]: even if cpld_control_noaero_p8_gnu doesn't show same OSC pt2pt issue, we will not be able to pin-point out gnu/openmpi issue. Also, it doesn't explain anything about control_wam_debug_gnu failure issue. We should revisit when new version installation of gnu is available. Of course, gocart might be sensitive with specific version of mpich or openmpi. @zach1221 I think we should move on for merging.

Comment (jkbk2004) [reply tests/rt.conf:416]: I confirm cpld_control_noaero_p8_gnu is running ok on hera/rocky8.

**Review by jkbk2004 — APPROVED** [review]

**Review by zach1221 — APPROVED** [review]

**Comment by jkbk2004** [comment]
@aerorahul We moved to rocky8. FYI: we will revisit about the gnu/openmpi issue on rocky8.

**Pull request merged**